### PR TITLE
Portfolio details popup window

### PR DESCRIPTION
**Pull request by EddyShimwa**
### We added a couple of changes in this pull request and some of them are:
- At first, we created a new branch for this project.
- Then we added popup details functionality in javascript.
- Then we added the necessary CSS styles.
- Last but not least we fixed the linter errors.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -11,4 +11,152 @@ hamburger.addEventListener('click', toggleNav);
...
+  popContainer.append(popupModal);
+}
+
+const actionsBtn = document.querySelectorAll('.see-more-action');
+actionsBtn.forEach((btn) => {
+  btn.addEventListener('click', (e) => {
+    const { id } = e.target;
+    fetchOneProject(id);
+  });
+});
```
Comment (BregornOriginal) [inline app.js:162]: Good job creating an addEventListener.

- [ ] Please make the 'See Project' button works correctly, when I click on the button nothing happens. As this is a requirement of the project
- [ ] Remember to make the 'x' button close the popup window and it returns the user to the same place that he was before joining the project

image

File: app.js [hunk 2]
```diff
@@ -11,4 +11,152 @@ hamburger.addEventListener('click', toggleNav);
...
+    title: 'Multi-Post Stories',
+    frame: ['FACEBOOK', 'FULL STACK DEV', 2015],
+    tags: ['HTML', 'CSS', 'Javascript'],
+    imageUrl: './images/feature-img4.png',
+    projectDetails:
+      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the releaLorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the releorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum han printer took a galley of type and scrambled it 1960s with the releawn printer took a galley of type and scrambled it 1960s with the releaLorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the relea",
+    liveLink: '#',
+    sourceLink: '#',
+  },
+];
```
Comment (BregornOriginal) [inline app.js:62]: Good job creating an object to contain the different projects. It's would be very helpful in your future when you add the projects that you will create on the following modules! 💯 💯

**Review by BregornOriginal — CHANGES_REQUESTED** [review]
Hi @EddyShimwa , 

## Good job so far! 🥇 🥇 🥇
There are some issues that you still need to work on to go to the next project but you are almost there!

- [ ] Kindly UPDATE your README file with the information of your partner that's contributed to this repository ❗ ❗ ❗ 

## Highlights

✅ Linters are passing
✅ Correct use of GitHub Flow
✅ Good commit history
✅ Good work collaborating with your team! 
✅ Good Pull Request description

one-step-closer-one-more

## Required Changes ♻️ ♻️ ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.js (hunk 1 above)
Comment (raihan2bd) [reply app.js:162]: Hi @BregornOriginal, Thanks for your valuable feedback But I checked our See Project button and it is working correctly. Can you elaborate on my javascript code problem clearly?

Comment (BregornOriginal) [reply app.js:162]: @raihan2bd You're right, let me perform this review, please. It seems that my browser doesn't work properly early

**Review by BregornOriginal — CHANGES_REQUESTED** [review]
Hi @EddyShimwa , 

## Good job so far! 🥇 🥇 🥇
There are some issues that you still need to work on to go to the next project but you are almost there!

- [ ] Kindly UPDATE your README file with the information of your partner that's contributed to this repository ❗ ❗ ❗ 

## Highlights

✅ Linters are passing
✅ Correct use of GitHub Flow
✅ Good commit history
✅ Good work collaborating with your team! 
✅ Good Pull Request description

one-step-closer-one-more

## Required Changes ♻️ ♻️ ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html [hunk 3]
```diff
@@ -89,7 +89,7 @@ <h1>Tonic</h1>
               <li>CSS</li>
               <li>javaScript</li>
             </ul>
-            <span>See Project</span>
+            <span class="see-more-action" id="project1">See Project</span>
           </div>
         </div>
       </section>
```
Comment (Kamba56) [inline index.html:95]: - [Optional] Please make sure that the project section is loaded dynamically from the js file. You should not have any project here, instead, your project array in your js file should be used to populate this project section. This is part of the project requirements. Kindly fix this.

File: app.js [hunk 4]
```diff
@@ -11,4 +11,152 @@ hamburger.addEventListener('click', toggleNav);
...
+
+  popContainer.append(popupModal);
+}
+
+const actionsBtn = document.querySelectorAll('.see-more-action');
+actionsBtn.forEach((btn) => {
+  btn.addEventListener('click', (e) => {
+    const { id } = e.target;
+    fetchOneProject(id);
+  });
```
Comment (Kamba56) [inline app.js:161]: - [ ] Your popup isn't displaying at all. Kindly revisit your code and fix this, please.

- [ ] As requested by the previous reviewer, consider adding your partner's details to your readme file.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @EddyShimwa , 

Great work on making the changes requested by a previous reviewer 👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- [x] Great use of Github workflow!👏👏👏
- [x] All linters passed!👏👏👏

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/app.js [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+  const actBtn = document.createElement('a');
+  actBtn.classList.add('act-btn');
+  actBtn.setAttribute('href', project.liveLink);
+  actBtn.innerHTML = "See Live <img src='./images/live-link.png' alt='live-link'>";
+  actions.appendChild(actBtn);
+
+  const actBtnSource = document.createElement('a');
+  actBtnSource.classList.add('act-btn');
+  actBtnSource.setAttribute('href', project.sourceLink);
```
Comment (sajid-munawar) [inline js/app.js:142]: - [OPTIONAL] Kindly add  a link to the live link button and a link to sourcelink button.

**Review by sajid-munawar — APPROVED** [review]
### Hi @EddyShimwa , 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Followed Github flow :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and description :heavy_check_mark:
 
 ## [Optional] suggestions:
 * [ ]  Please check my inline comments.

 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**